src/main/java/org/springframework/data/elasticsearch/core/DocumentOperations.java

```diff
@@ -113,6 +123,24 @@
 	 */
 	<T> List<T> multiGet(Query query, Class<T> clazz, IndexCoordinates index);
 
+	/**
+	 * Check if an entity with given {@literal id} exists.
+	 *
+	 * @param id the {@literal _id} of the document to look for.
+	 * @param clazz the domain type used.
+	 * @return {@literal true} if a matching document exists, {@literal false} otherwise.
+	 */
+	<T> Boolean exists(String id, Class<T> clazz);
```

Comment:
Nit: Use primitive `boolean` to avoid potential null issues. Also, we do not need `T` here. `Class<?>` is fine.